### DISCUSSION TITLE
Delete the job after the specified time has passed after the job is completed

### DISCUSSION THREAD

**Discussion by DamonWz** [quartznet/quartznet#1226] [Q&A]
Hi! I`m using Quartz 2.3 in springboot application and mysql to be the job store. 
So how to delete job automatically after a specified time has passed after the job is completed? Such as in 7 days, or in 1 month.
By the way that I have used the annotation @PersistJobDataAfterExecution in my own job class.
Thank you advance! And sorry for my English.

**Comment by Sicos1977** [comment]
Just make another job that deletes finished jobs

**Reply by DamonWz** [reply]
Thank you for your reply! 
I need to create another job class who`s responsible for deleting the original job and store it like you said, so they`ll take twice as much of space. Is there any configuration items of quartz, like survival time, so that the job will disappear after the configuration time has passed?

**Reply by lahma** [reply]
Quartz will delete jobs that have no more triggers referring to them, but there's no configurable wait time before deletion. Jobs will stay in DB if job is marked to be durable even if there are no triggers referring to it.

P.S. this is the repo for .NET version of Quartz, but the logic is the same.

**Reply by DamonWz** [reply]
Got it! Thanks!
